components/camel-undertow/src/main/java/org/apache/camel/component/undertow/UndertowComponent.java

```diff
@@ -306,6 +339,8 @@ public Producer createProducer(CamelContext camelContext, String host,
     protected void doStart() throws Exception {
         super.doStart();
 
+        initSecurityProvider();
```

Comment:
Only init if the security provider is null

Reply:
I'll change it